api/resources_portal/views/attachment.py

```diff
@@ -21,45 +22,37 @@ class Meta:
             "created_at",
             "updated_at",
             "sequence_map_for",
+            "owned_by_org",
+            "owned_by_user",
         )
         read_only_fields = ("id", "created_at", "updated_at")
 
 
-def user_has_perm_on_active_material_request(user, perm):
+class AttachmentDetailSerializer(AttachmentSerializer):
+    sequence_map_for = MaterialRelationSerializer(many=False, read_only=True)
+    owned_by_org = OrganizationRelationSerializer(many=False)
+    owned_by_user = UserRelationSerializer(many=False)
 
-    # Retrieve all organization permissions in a single query
-    checker = ObjectPermissionChecker(user)
-    organizations = Organization.objects.all()
-    checker.prefetch_perms(organizations)
 
+def user_is_requester_on_active_material_request(user, organization):
     # Uses prefetch_related to retrieve all related objects in a single query
-    for organization in user.organizations.all().prefetch_related("materials"):
-        if checker.has_perm(perm, organization):
-            for material in organization.materials.all().prefetch_related("requests"):
-                for request in material.requests.all():
-                    if request.is_active:
-                        return True
+    for material in organization.materials.all().prefetch_related("requests"):
+        for material_request in material.requests.all():
+            if material_request.is_active and material_request.requester == user:
+                return True
     return False
 
 
-class AttachmentDetailSerializer(AttachmentSerializer):
-    sequence_map_for = MaterialRelationSerializer(many=False, read_only=True)
-
-
-class CanViewRequestsOrIsRequesterOrIsAdminUser(BasePermission):
+class OwnsAttachmentOrIsAdmin(BasePermission):
     def has_object_permission(self, request, view, obj):
-        return (
-            MaterialRequest.objects.filter(requester=request.user, is_active=True).exists()
-            or user_has_perm_on_active_material_request(request.user, "view_requests")
-            or request.user.is_staff
-        )
+        return request.user in obj.owned_by_org.members.all() or request.user.is_staff
 
 
-class CanApproveRequestsOrIsRequesterOrIsAdminUser(BasePermission):
+class OwnsAttachmentOrIsRequesterOrIsAdmin(BasePermission):
     def has_object_permission(self, request, view, obj):
         return (
-            MaterialRequest.objects.filter(requester=request.user, is_active=True).exists()
-            or user_has_perm_on_active_material_request(request.user, "approve_requests")
+            request.user in obj.owned_by_org.members.all()
+            or user_is_requester_on_active_material_request(request.user, obj.owned_by_org)
```

Comment:
I think we can use `request.user == obj.owned_by_user or request.user in obj.owned_by_org.members.all() or request.user.is_staff` and call it `OwnsAttachmentOrIsAdmin`, and the we don't need to worry about whether the user is the requester or not, and can nix the `user_is_requester_on_active_material_request` function.